packages/design-picker/src/components/unified-design-picker.tsx

```diff
@@ -205,7 +205,13 @@ const DesignButton: React.FC< DesignButtonProps > = ( {
 						) }
 						{ isEnableThemeStyleVariations && style_variations.length > 0 && (
 							<div className="design-picker__options-style-variations">
-								<StyleVariationBadges variations={ style_variations } />
+								<StyleVariationBadges
+									variations={ style_variations }
+									onClick={ ( variation ) => {
+										design.active_style_variation_slug = variation.slug;
```

Comment:
Could you check my PR here: https://github.com/Automattic/wp-calypso/pull/67504? I added an action to set the selected style variation to the store, so maybe you could reuse that?

There might be some technicalities. For example at this point the style variation is missing `inline_css`, so we need to "re-select" the variation with the inline css after we fetch the design details.

Reply:
Sure, let me check 👀